plugins/inputs/win_wmi/win_wmi.go

```diff
@@ -0,0 +1,265 @@
+//go:build windows
+// +build windows
+
+package win_wmi
+
+import (
+	_ "embed"
+	"fmt"
+	"runtime"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	ole "github.com/go-ole/go-ole"
+	"github.com/go-ole/go-ole/oleutil"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/filter"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+// Query struct
+type Query struct {
+	query                string
+	Namespace            string   `toml:"namespace"`
+	ClassName            string   `toml:"class_name"`
+	Properties           []string `toml:"properties"`
+	Filter               string   `toml:"filter"`
+	TagPropertiesInclude []string `toml:"tag_properties"`
+	tagFilter            filter.Filter
+}
+
+// Wmi struct
+type Wmi struct {
+	Queries []Query `toml:"query"`
+	Log     telegraf.Logger
+}
+
+// S_FALSE is returned by CoInitializeEx if it was already called on this thread.
+const sFalse = 0x00000001
+
+func oleInt64(item *ole.IDispatch, prop string) (int64, error) {
+	v, err := oleutil.GetProperty(item, prop)
+	if err != nil {
+		return 0, err
+	}
+	defer func() { _ = v.Clear() }()
+
+	i := v.Val
+	return i, nil
+}
+
+// Init function
+func (s *Wmi) Init() error {
+	if err := CompileInputs(s); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Description function
+func (s *Wmi) Description() string {
+	return "returns WMI query results as metrics"
+}
+
```

Comment:
We do not use the Description function anymore.

```suggestion
```

Reply:
When I try to build after removing the Description function, I get:
```
plugins/inputs/win_wmi/win_wmi.go:263:55: cannot use &Wmi{} (value of type *Wmi) as type telegraf.Input in return statement:
        *Wmi does not implement telegraf.Input (missing Description method)
```

Reply:
Here is my attempt from your branch:

```
❯ git diff
diff --git a/plugins/inputs/win_wmi/win_wmi.go b/plugins/inputs/win_wmi/win_wmi.go
index ac78e7516..8f5563b31 100644
--- a/plugins/inputs/win_wmi/win_wmi.go
+++ b/plugins/inputs/win_wmi/win_wmi.go
@@ -62,11 +62,6 @@ func (s *Wmi) Init() error {
        return nil
 }
 
-// Description function
-func (s *Wmi) Description() string {
-       return "returns WMI query results as metrics"
-}
-
 // SampleConfig function
 func (s *Wmi) SampleConfig() string {
        return sampleConfig
❯ GOOS=windows make
go mod download -x
env -u GOOS -u GOARCH -u GOARM -- go build -o ./tools/custom_builder/custom_builder ./tools/custom_builder
env -u GOOS -u GOARCH -u GOARM -- go build -o ./tools/license_checker/license_checker ./tools/license_checker
env -u GOOS -u GOARCH -u GOARM -- go build -o ./tools/readme_config_includer/generator ./tools/readme_config_includer/generator.go
go generate -run="readme_config_includer/generator$" ./plugins/inputs/...
go generate -run="readme_config_includer/generator$" ./plugins/outputs/...
go generate -run="readme_config_includer/generator$" ./plugins/processors/...
go generate -run="readme_config_includer/generator$" ./plugins/aggregators/...
go generate -run="readme_config_includer/generator$" ./plugins/secretstores/...
go build -tags "" -ldflags " -X github.com/influxdata/telegraf/internal.Commit=9fbe21a6 -X github.com/influxdata/telegraf/internal.Branch=master -X github.com/influxdata/telegraf/internal.Version=1.26.0-9fbe21a6" ./cmd/telegraf
```

Reply:
Removed.